src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -121,4 +121,22 @@ public static Tag parseTag(String tag) throws ParseException {
         }
         return tagSet;
     }
+
+    /**
+     * Parses {@code oneBasedIndex} into an Integer and returns it. Leading and trailing whitespaces will be
+     * trimmed.
+     * @throws ParseException if the specified index is invalid (not non-zero unsigned integer).
+     */
+    public static Integer parseNumber(String oneBasedIndex) throws ParseException {
+        String trimmedIndex = oneBasedIndex.trim();
+        if (!StringUtil.isNonZeroUnsignedInteger(trimmedIndex)) {
+            throw new ParseException(MESSAGE_INVALID_INDEX);
+        }
+
+        if (!StringUtil.isInvalidNumber(trimmedIndex)) {
+            throw new ParseException(MESSAGE_INVALID_INDEX);
+        }
+
+        return Integer.parseInt(trimmedIndex.replaceAll("[^0-9]", ""));
```

Comment:
What if the user typed something like `delt b3`, should the command run or should it prompt an error?

Reply:
Also is it necessary to call `replaceAll()` here? I think when you check for `isNonZeroUnsignedInteger` and `isInvalidNumber`, you already assured that the `trimmedIndex` is a valid number.

Reply:
Fixed with update